utility/src/test/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsAssessorImplTest.kt

```diff
@@ -0,0 +1,281 @@
+import org.robolectric.Shadows.shadowOf
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import org.robolectric.shadows.ShadowActivityManager
+import java.io.File
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val TEST_PACKAGE_LABEL = "TEST_PACKAGE_LABEL"
+private const val TEST_APP_PATH = "TEST_APP_PATH"
+private const val TEST_PID = 1
+private const val TEST_FILE_NAME = "TEST_FILE_NAME"
+private const val TEST_FILE_CONTENT = "TEST_FILE_CONTENT"
+
+/** Tests for [PerformanceMetricsAssessorImpl]. */
+// FunctionName: test names are conventionally named with underscores.
+@Suppress("FunctionName")
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = PerformanceMetricsAssessorImplTest.TestApplication::class)
+class PerformanceMetricsAssessorImplTest {
+
+  @Inject
+  lateinit var performanceMetricsUtilsImpl: PerformanceMetricsAssessorImpl
+
+  @Inject
+  lateinit var context: Context
+
+  @Before
+  fun setUp() {
+    MockitoAnnotations.initMocks(this)
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_setTotalMemory_returnsCorrectMemoryTier() {
+    val activityManager: ActivityManager =
+      ApplicationProvider.getApplicationContext<Application>()
+        .getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+    val shadowActivityManager: ShadowActivityManager = shadowOf(activityManager)
+    val memoryInfo = ActivityManager.MemoryInfo()
+    memoryInfo.totalMem = (1.5 * 1024 * 1024 * 1024).toLong()
+    shadowActivityManager.setMemoryInfo(memoryInfo)
+    val memoryTier = performanceMetricsUtilsImpl.getDeviceMemoryTier()
+
+    assertThat(memoryTier).isEqualTo(OppiaMetricLog.MemoryTier.LOW_MEMORY_TIER)
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_getTotalStorageUsed_returnsCorrectStorageUsage() {
+    context.openFileOutput(TEST_FILE_NAME, Context.MODE_PRIVATE).use {
+      it.write(TEST_FILE_CONTENT.toByteArray())
+    }
+    val cacheFile = File.createTempFile(TEST_APP_PATH, null, context.cacheDir)
+    val expectedStorageValue = TEST_FILE_CONTENT.length + cacheFile.length()
+
+    assertThat(performanceMetricsUtilsImpl.getUsedStorage()).isEqualTo(expectedStorageValue)
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_getTotalStorageUsageTier_returnsCorrectStorageUsageTier() {
+    context.openFileOutput(TEST_FILE_NAME, Context.MODE_PRIVATE).use {
+      it.write(TEST_FILE_CONTENT.toByteArray())
+    }
+    val cacheFile = File.createTempFile(TEST_APP_PATH, null, context.cacheDir)
+    val expectedStorageValue = TEST_FILE_CONTENT.length + cacheFile.length()
+
+    val expectedStorageTierValue = when (expectedStorageValue / (1024 * 1024 * 1024)) {
+      in 0L until 32L -> OppiaMetricLog.StorageTier.LOW_STORAGE
+      in 32L..64L -> OppiaMetricLog.StorageTier.MEDIUM_STORAGE
+      else -> OppiaMetricLog.StorageTier.HIGH_STORAGE
+    }
```

Comment:
The test should be verifying an exact state, not computing it (prefer no control flow or logic in tests where we can help it).

Reply:
modified.